lib/media/streaming_engine.js

```diff
@@ -674,7 +674,10 @@ shaka.media.StreamingEngine = class {
     // that since we just switched.
     const newSegment = this.getSegmentReferenceNeeded_(
         mediaState, presentationTime, bufferEnd, periodIndex);
-    let newSegmentSize = newSegment ? newSegment.getSize() : null;
+    let newSegmentSize =
+      (newSegment ? newSegment.getSize() : null) ||
+      mediaState.operation.getBytesTotal() ||
```

Comment:
I don't see why getBytesTotal() of the in-progress segment request should be considered part of the calculation of the new segment size.  This doesn't seem to improve the original problem, which was that you don't know the size of the new segment.